packages/lake/src/components/LakeTagInput.tsx

```diff
@@ -0,0 +1,219 @@
+    paddingTop: spacings[4],
+  },
+});
+
+export type LakeTagInputProps = Merge<
+  TextInputProps,
+  {
+    readOnly?: boolean;
+    error?: string;
+    disabled?: boolean;
+    valid?: boolean;
+    hideErrors?: boolean;
+    help?: string;
+    validator?: (value: string) => boolean;
+    values: string[];
+    onAddValues: (added: string[]) => void;
+    onRemoveValue: (value: string) => void;
+  }
+>;
+
+const SEPARATORS_REGEX = /,| /;
+
+export const LakeTagInput = forwardRef<TextInput | null, LakeTagInputProps>(
+  (
+    {
+      validator = () => true,
+      onFocus: originalOnFocus,
+      onBlur: originalOnBlur,
+      values,
+      onAddValues,
+      onRemoveValue,
+      readOnly = false,
+      disabled = false,
+      valid = false,
+      hideErrors = false,
+      help,
+      error,
+    }: LakeTagInputProps,
+    forwardRef,
+  ) => {
+    const inputRef = useRef<TextInput | null>(null);
+    const containerRef = useRef<View | null>(null);
+    const [isFocused, setIsFocused] = useState(false);
+    const [isHovered, setIsHovered] = useState(false);
+
+    useHover(containerRef, {
+      onHoverStart: () => setIsHovered(true),
+      onHoverEnd: () => setIsHovered(false),
+    });
+
+    const onTextInputChange = useCallback((value: string) => {
+      const values = [...new Set(value.split(SEPARATORS_REGEX).filter(s => s.length))];
+      if (values.length > 1 || values[0] !== value) {
+        onAddValues(values);
+        inputRef.current?.clear();
+      }
+    }, []);
+
+    const autoFocus = useCallback(() => {
+      inputRef.current?.focus();
+    }, []);
+
+    const onFocus = useCallback(
+      (event: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        setIsFocused(true);
+        originalOnFocus?.(event);
+      },
+      [originalOnFocus],
+    );
+
+    const onBlur = useCallback(
+      (event: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        setIsFocused(false);
+        originalOnBlur?.(event);
+      },
+      [originalOnBlur],
+    );
+
+    const mergedRef = useMergeRefs(inputRef, forwardRef);
+    const hasError = isNotNullishOrEmpty(error);
+
+    return (
+      <View>
+        <Pressable
+          style={[
+            styles.root,
+            readOnly && hasError && styles.readOnlyError,
+            disabled && styles.disabled,
+            readOnly && styles.readOnly,
+            isFocused && styles.focused,
+            hasError && styles.error,
+            valid && styles.valid,
+            isHovered && styles.hovered,
+          ]}
+          aria-errormessage={error}
+          onPress={autoFocus}
+          ref={containerRef}
+        >
+          {values.map((v, i) => (
+            <Tag
+              key={i}
+              onPressRemove={!readOnly && !disabled ? () => onRemoveValue(v) : undefined}
+              style={styles.tag}
+              color={validator(v) ? "gray" : "negative"}
+            >
+              {v}
+            </Tag>
+          ))}
```

Comment:
<img width="474" alt="Screenshot 2023-12-15 at 14 47 01" src="https://github.com/swan-io/lake/assets/1688645/669cea83-91d8-4d73-b91b-59292e0faaf1">

tags can overflow, shouldn't we give them a maxWidth?